### PR TITLE
Add support for custom fields in channels

### PR DESCRIPTION
**Pull request by Engerim**
* support for custom fields in channels
* add tests for this. tests are deactivate for the mock server as the mock server can not handle key reference

### CODE REVIEW

File: commercetools/resource_channel_test.go [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+	 depends_on = [
+		commercetools_type.channel_test,
+	  ]
+	roles = ["Primary"]
+	key  = "standard-key"
+ 	custom {
+		type_key = "channel-test"
+
+		field {
+		  key = "carrier"
```
Comment (mvantellingen) [inline commercetools/resource_channel_test.go:301]: should this perhaps be named `name = "carrier"` to be consistent?

**Comment by mvantellingen** [comment]
Awesome work; one minor question added. I think this is the first resource with support for setting custom fields directly

File: commercetools/resource_channel_test.go (hunk 1 above)
Comment (Engerim) [reply commercetools/resource_channel_test.go:301]: can change it also to name, I don't have any strong feelings about it. Will change it tomorrow

**Comment by fpieper-diva-e** [comment]
Hi @Engerim @mvantellingen ,
do you guys have any idea on when this might be mergend and released? We would be very interested in using this!
Cheers,
Felix

**Comment by Engerim** [comment]
Hi, I have no clue I’m not able to merge it what we did is using our fork to use this feature for us already

**Comment by Engerim** [comment]
I have updated the pr with the latest changes evtl it is still possible to merge it @mvantellingen

**Comment by mvantellingen** [comment]
Follow up on #265

**Comment by mvantellingen** [comment]
Thanks @Engerim. I wanted to continue working on this with your branch but decided to start over with a new approach. I'll make sure you are mentioned in the release notes for your work

**Pull request closed without merging**